apps/core/app/[locale]/(default)/account/register-customer/page.tsx

```diff
@@ -0,0 +1,86 @@
+import { redirect } from 'next/navigation';
+import { NextIntlClientProvider } from 'next-intl';
+import { getMessages, getTranslations } from 'next-intl/server';
+
+import { auth } from '~/auth';
+import { FormFieldSortInput } from '~/client/generated/graphql';
+import { getCountries } from '~/client/management/get-countries';
+import { getAddressFormFields } from '~/client/queries/get-address-form-fields';
+import { getCustomerFormFields } from '~/client/queries/get-customer-form-fields';
+import { getReCaptchaSettings } from '~/client/queries/get-recaptcha-settings';
+import { getStoreCountry } from '~/client/queries/get-store-country';
```

Comment:
Can we do this with a single query?

Reply:
Interesting idea, but `getAddressFormFields`, `getAddressFormFields` and `getReCaptchaSettings` need to be used in separate cases 🤔

Reply:
You can try something like what @deini is doing with [fragment colocation](https://github.com/bigcommerce/catalyst/pull/766). What do you think?

Reply:
@jorgemoya thats great idea, thanks. Refactored according to the new approach 👍